lib/modules/platform/local/scm.ts

```diff
@@ -0,0 +1,52 @@
+import { execSync } from 'node:child_process';
+import { glob } from 'glob';
+import { logger } from '../../../logger';
+import type { CommitFilesConfig, CommitSha } from '../../../util/git/types';
+import type { PlatformScm } from '../types';
+
+let fileList: string[] | undefined;
+export class LocalFs implements PlatformScm {
+
+  isBranchBehindBase(branchName: string, baseBranch: string): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+  isBranchModified(branchName: string): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+  isBranchConflicted(baseBranch: string, branch: string): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+  branchExists(branchName: string): Promise<boolean> {
+    return Promise.resolve(true);
+  }
+  getBranchCommit(branchName: string): Promise<string | null> {
+    return Promise.resolve(null);
+  }
+  deleteBranch(branchName: string): Promise<void> {
+    return Promise.resolve();
+  }
+  commitAndPush(commitConfig: CommitFilesConfig): Promise<string | null> {
+    return Promise.resolve(null);
+  }
+
+  async getFileList(): Promise<string[]> {
+    try {
+      // fetch file list using git
+      const stdout = execSync('git ls-files', { encoding: 'utf-8' });
```

Comment:
why execSync when we can be async here?

Reply:
1. I couldn't get the mocking working when it was promisified exec
2. I didn't think it matters for this use case as we block on this getFileList() anyway

Reply:
😏